scripts/monitoring/cron-send-logging-checks.py

```diff
@@ -0,0 +1,224 @@
+                    else:
+                        es_status['pods'][pod_dc]['elasticsearch_health'] = 0
+
+                    # Compare the master across all ES nodes to see if we have split brain
+                    curl_cmd = "{} 'https://localhost:9200/_cat/master'".format(self.es_curl)
+                    es_master = "exec -ti {} -- {}".format(pod_name, curl_cmd)
+                    master_name = self.oc.run_user_cmd(es_master).split(' ')[1]
+
+                    if es_status['single_master'] == None:
+                        es_status['single_master'] = 1
+                        es_master_name = master_name
+                    elif es_master_name != master_name:
+                        es_status['single_master'] = 0
+
+        # get cluster nodes
+        curl_cmd = "{} 'https://localhost:9200/_nodes'".format(self.es_curl)
+        node_cmd = "exec -ti {} -- {}".format(es_master_name, curl_cmd)
+        cluster_nodes = self.oc.run_user_cmd(node_cmd)['nodes']
+        es_status['all_nodes_registered'] = 1
+        # The internal ES node name is a random string we do not track anywhere
+        # pylint: disable=unused-variable
+        for node, data in cluster_nodes.items():
+        # pylint: enable=unused-variable
+            has_matched = False
+            for pod in es_pods:
+                if data['host'] == pod['metadata']['name']:
+                    has_matched = True
+
+            if has_matched == False:
+                es_status['all_nodes_registered'] = 0
+
+        return es_status
+
+    def check_fluentd(self):
+        ''' Verify fluentd is running '''
+        fluentd_status = {}
+
+        # Get all nodes with fluentd label
+        nodes = self.oc.get_nodes()
+        fluentd_nodes = []
+        for node in nodes['items']:
+            if 'logging-infra-fluentd' in node['metadata']['labels']:
+                if node['metadata']['labels']['logging-infra-fluentd'] == 'true':
+                    fluentd_nodes.append(node)
+
+        # Get all fluentd pods
+        pods = self.oc.get_pods()
+        fluentd_pods = []
+        for pod in pods['items']:
+            if 'component' in pod['metadata']['labels']:
+                if pod['metadata']['labels']['component'] == 'fluentd':
+                    fluentd_pods.append(pod)
+
+        # Make sure fluentd is on all the nodes and the pods are running
+        fluentd_status['number_expected_pods'] = len(fluentd_nodes)
+        fluentd_status['number_pods'] = len(fluentd_pods)
+        fluentd_status['node_mismatch'] = 0
+        for node in fluentd_nodes:
+            node_matched = False
+            for pod in fluentd_pods:
+                if node['metadata']['labels']['kubernetes.io/hostname'] == pod['spec']['host']:
+                    node_matched = True
+
+            if node_matched == False:
+                fluentd_status['node_mismatch'] = 1
+
+        fluentd_status['running'] = 1
+        for pod in fluentd_pods:
```

Comment:
By the time this loop is done, we appear to have iterated the nodes twice, and the pods twice, and then iterated the fluentd pods for each node again.

Perhaps we can consider a way to avoid the number of iterations over the data.